### PR TITLE
Add comment on unchecked arithmetic (division by zero) in `Math.sol`

### PR DESCRIPTION
**Pull request by pcaversaccio**
As title. Let me know if this requires a Changeset entry.

### CODE REVIEW

**Comment by Amxx** [comment]
Interesting to note that the div opcode will return 0 when dividing by 0. Using assembly without a "manual" check would be incorrect.

**Comment by pcaversaccio** [comment]
After a quick chat with @Amxx I would like to quickly explain why it's worth documenting. When I first looked at the code I was concerned about where the revert comes from because of the `unchecked` block and the usage of assembly? So the underlying reason is that the `DIV` opcode returns 0 for a denominator of 0 (hello footgun :)). That's why Uniswap, for example, uses a [check](https://github.com/Uniswap/v3-core/blob/412d9b236a1e75a98568d49b1aeb21e3a1430544/contracts/libraries/FullMath.sol#L34) beforehand:

image

Maybe it's just a personal thing, but imho it's worth highlighting that Solidity does add here an additional check.

File: contracts/utils/math/Math.sol [hunk 1]
```diff
@@ -67,6 +67,8 @@ library Math {
 
             // Handle non-overflow cases, 256 by 256 division.
             if (prod1 == 0) {
+                // Will revert if denominator == 0. It is not possible to disable the check for division by zero using the unchecked block.
```
Comment (frangio) [inline contracts/utils/math/Math.sol:70]: I think this expresses the message better:

```suggestion
                // Will revert if denominator == 0. The surrounding unchecked block does not change this fact.
```

Comment (pcaversaccio) [reply contracts/utils/math/Math.sol:70]: Agree with the wording change.

**Review by frangio — DISMISSED** [review]
LGTM. Thanks!

**Comment by frangio** [comment]
Expanded the wording based on feedback from @Amxx.

**Review by Amxx — APPROVED** [review]

**Pull request merged**